core/root/config/templates/cluster.yaml

```diff
@@ -1189,29 +1159,10 @@ kubernetes:
 # Kubernetes Self-hosted networking daemonsets
 # Choose either 'canal' (calico+flannel) or 'flannel'
 # (choose 'canal' if you require calico or kubernetes NetworkPolicy firewalling).
-# When enabled nodes no longer need to connect etcd and frees up node podCIDR leases faster resolving issue:
-# https://github.com/coreos/flannel/issues/954.
-#
 # Enable Typha if you have 50+ nodes in your cluster. Without Typha, the load on the API server and Felix’s CPU usage
-# increases substantially as the number of nodes is increased.  The legacy top-level 'useCalico' setting has no effect with selfHosting
-# enabled - use type 'canal' for calico+flannel.
-#
-# Notes on migrating between networking types: -
-# From Legacy Calico to self-hosted Canal - GOOD
-# From Legacy Calico to self-hosted Flannel - OK - ISSUES1
-# From self-hosted Flannel to self-hosted Canal - GOOD
-# From self-hosted Canal to self-hosted Flannel - OK - ISSUES1
-# From Legacy Flannel to self-hosted Canal or Flannel - ISSUES2
-# From self-hosted Canal or Flannel to legacy Flannel - ISSUES3
-#
-# ISSUES1 - calico iptables rules remain on the nodes, recycle nodes to remove.
-# ISSUES2 - Existing pod ip addresses are not tracked and can cause clashing ips - please cordon all nodes whilst
-# performing the upgrade until all workers/nodes have been recycyled - then it will be ok.
-# ISSUES3 - Rolling back to old networking will require all nodes are cycled/rolled and the network is disrupted throughout.
-
+# increases substantially as the number of nodes is increased.
   networking:
     selfHosting:
-      enabled: true    # false will fall back to legacy coreos flannel/etcd2 installation
       type: canal      # either "canal" or "flannel"
       typha: false     # enable for type 'canal' for 50+ node clusters
```

Comment:
Do we have any reasons not to enable typha by default btw?

Reply:
Typha is only needed on larger clusters with 50+ nodes.

Reply:
Would the purpose of avoiding it when unnecessary is to avoid moving parts?
I just wondered if it would be a good idea to enable it by default if it doesn't have any bad side-effect, so that users won't suffer from degraded performance after reaching 50+ nodes

Reply:
I'm a little worried about trying to set it automatically because that would mean I would have to count the nodes in the nodepools and make allowances for possible cluster auto-scaling.  With it off by default, a cluster admin with a growing cluster and seeing high loads on their apiservers will hopefully have had some time to start getting acquainted with the cluster.yaml file and settings.

Reply:
Makes sense. Thanks a lot for clarifying!